### ISSUE TITLE
NIVEASHOP - compliance issues

### ISSUE THREAD

**Issue by BLR-0118** [ONDC-Official/Pre-production#233]
Logs with mystore:
1. All callback APIs - core_version should be "1.0.0", value <= maximum_value, message.context is invalid;
2. /on_select - item attributes which show in the catalog are not part of the API contract for /on_select;
3. /on_select - quote.price should be 255+75 = 330 (shows correctly in /on_init though);
4. /init - fulfillments.end.location.area_code is not same as in /select even though the gps coords are same. Why?
5. /on_init - same issue as /on_search with core_version, message.context; billing details aren't part of API contract;
6. /on_confirm - order.id different from /confirm; if order auto-accepted, state should be "Accepted";
7. /on_status - quote.price is 405 here. why?
8. No APIs for cancel, update, support? (understand that you're items are non-returnable/non-cancellable, hence /update may not be relevant, but what about cancel, support APIs)?

**Comment by BLR-0118** [comment]
@Agam1997

**Comment by Agam1997** [comment]
Hey @BLR-0118 
/on_init - same issue as /on_search with core_version, message.context; billing details aren't part of API contract; - on_init shares billing details as per the document we have. 

We do not support order cancellation once placed. 
Nivea has an ORM team that handles all support-related queries for orders and the support contact is shared on the /on_search.

**Comment by BLR-0118** [comment]
All issues mentioned yesterday have to be fixed and logs resubmitted for verification.

**Comment by BLR-0118** [comment]
@Agam1997

**Comment by Agam1997** [comment]
@BLR-0118 All the issues mentioned are fixed in the latest pull request.

**Comment by BLR-0118** [comment]
1. /search has fulfillment end location in Mumbai and /on_search has provider location in Bilaspur. However, /on_select shows fulfillment TAT of 5 mins for which there's a delivery charge of 75. How?
2. /on_confirm - quote.price is 330 but payment.params.amount is "NaN". Also, payment.params.transaction_id which shows in /confirm isn't shown here. Why?
3. /on_status - for same order, /on_confirm shows price of 330 but here it's 405. Why?
4. No APIs for support, cancel, update, track.

**Comment by BLR-0118** [comment]
@Agam1997

**Comment by BLR-0118** [comment]
Pls also check your compliance with the checklist [here](https://docs.google.com/document/d/1jXkPwF-Fol9LeXbzQaC65R6UB60Cwc-cUiQ0qzsKNKQ/edit)

**Comment by BLR-0118** [comment]
Logs with mystore:
1. /on_select - ttl of "P24H" is invalid, should be "PT24H";
2. /on_init, /on_status - inconsistencies in attrs shown vs passed in /init. Additional attrs are reqd to be shown (as per API contract) in all these APIs;
3. fulfillment tracking is false in /on_confirm, yet how's tracking enabled?
4. no /support API?
5. what's your compliance with [this](https://docs.google.com/document/d/1jXkPwF-Fol9LeXbzQaC65R6UB60Cwc-cUiQ0qzsKNKQ/edit)

**Comment by BLR-0118** [comment]
@Agam1997

**Comment by BLR-0118** [comment]
1. /on_confirm - fulfillment state code is "pending" (lowercase) while in /on_status it's "Pending" (title case). Can you pls make this change?
2. Pls document your compliance with this checklist []([here](https://docs.google.com/document/d/1jXkPwF-Fol9LeXbzQaC65R6UB60Cwc-cUiQ0qzsKNKQ/edit))
Pls fix 1 & mention your response to 2 here. No need to resubmit logs.

**Comment by BLR-0118** [comment]
@Agam1997

**Comment by Agam1997** [comment]
1. Fixed from our side, Thanks for pointing it out.

- Search by item, category, provider & city is enabled 
- The brand has uploaded back images for packaged commodities, although we are also sending all details under different parameters of /on_search
- Buyer App finder fee amount is being absorbed in the seller end and currently not charging the customer. 
- No hardcoding of gateway end points. 
- Using Order.State & Fulfillment.State from provided links 
- We have tested error codes for business errors, item not available, Location Serviceability Error, Updation not possible, Cancellation not possible etc. and using codes as mentioned in the Error codes link
